core/deployment/src/main/java/io/quarkus/deployment/ConstructorPropertiesProcessor.java

```diff
@@ -0,0 +1,38 @@
+package io.quarkus.deployment;
+
+import java.beans.ConstructorProperties;
+
+import org.jboss.jandex.AnnotationInstance;
+import org.jboss.jandex.ClassInfo;
+import org.jboss.jandex.DotName;
+import org.jboss.jandex.MethodInfo;
+
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.CombinedIndexBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
+
+/**
+ * Registers all classes for reflection,
+ * that contain a constructor annotated with {@link ConstructorProperties}.
+ */
+public class ConstructorPropertiesProcessor {
+
+    private static final DotName CONSTRUCTOR_PROPERTIES = DotName.createSimple(ConstructorProperties.class.getName());
```

Comment:
Please use a String for the name instead. I prefer we don't have any dependency to `java.beans`.

Reply:
makes sense.